src/DBmysqlIterator.php

```diff
@@ -727,7 +727,7 @@ private function analyseFkey($values)
             } else if (count($values) == 3) {
                 $condition = array_pop($values);
                 $fkey = $this->analyseFkey($values);
-                $condition_value = $this->analyseCrit(current($condition));
```

Comment:
I do not understand this change. It breaks tests because it addss extra parenthesis on queries, but nothing more...
It may be for an untested case, it has to be added to the suite if so.